Tests/SentryTests/SentrySDKThreadTests.swift

```diff
@@ -0,0 +1,33 @@
+@testable import Sentry
+import XCTest
+
+final class SentrySDKThreadTests: XCTestCase {
+    func testRaceWhenBindingClient() {
+        for _ in 0..<10_000 {
```

Comment:
`m`: This test requires over a minute or so to run on my local computer. This will significantly increase the test duration. We don't need to use `SentrySDK.start` , which can be a heavy operation with side effects. Instead, we should only interact with the hub, cause that's where the problem lies.